### DISCUSSION TITLE
Run selection/line without execution of entire block

### DISCUSSION THREAD

**Discussion by skjerns** [posit-dev/positron#5055] [Q&A] [answered]
For debugging, I like to go line by line through my code by using the "Run selection/line in Console" feature (as I know from Spyder).

However, once in a while, not the current line, but the complete code block will be sent to the console. Is there a way to strictly only ever send the current line and not the code block to the console?

**Comment by juliasilge** [comment]
Can you share more about how/when you are seeing this? What is some example code where you see the whole code block sent to the console? Are you working in a `.py` file, `.qmd` file, or something else?

**Reply by skjerns** [reply]
It seems to happen whenever a nested loop is involved

```python
for x in range(10):
    print(x)  # here, run selection/line works
    print(x)
    print(x)
    for y in range(10):
        print(y)  # here, run selection/line runs the entire code block
        print(y)
        print(y)
```

**Reply by juliasilge** [reply, marked as answer]
Ah, I believe we are tracking this in https://github.com/posit-dev/positron/issues/3464. Take a look and see if you have any additional info to add! 🙌

**Reply by skjerns** [reply]
> Ah, I believe we are tracking this in #3464. Take a look and see if you have any additional info to add! 🙌

yes, that seems very similar!
